### PR TITLE
GetSchema rpc to streaming api

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR converts `GetSchema` RPC to a streaming api.

## Related Issue(s)

 - #11559 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [X] Tests were added or are not required
-   [X] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/schema/tracker.go [hunk 1]
```diff
@@ -134,21 +134,18 @@ func (t *Tracker) loadViews(conn queryservice.QueryService, target *querypb.Targ
...
+	// whenever a shard's primary tablet starts and sends the initial signal.
+	// This is needed clear out any stale view definitions.
 	t.clearKeyspaceViews(target.Keyspace)
-	t.updateViews(target.Keyspace, fvRes)
-	log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(fvRes))
 
-	return nil
+	return conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil, func(schemaRes *querypb.GetSchemaResponse) error {
+		t.updateViews(target.Keyspace, schemaRes.TableDefinition)
+		log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(schemaRes.TableDefinition))
```
Comment (GuptaManan100) [inline go/vt/vtgate/schema/tracker.go:146]: This log line is now incorrect. You can have the view definitions coming in separate packets, so you don't want to print 2 lines saying it finished loading the views for a keyspace and reporting different number of views each time, while the correct count is the sum of both of them

**Review by GuptaManan100 — APPROVED** [review]
Rest LGTM

File: go/vt/vtgate/schema/tracker.go (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vtgate/schema/tracker.go:146]: I thought about that. moving it out.

**Review by frouioui — APPROVED** [review]

File: go/vt/vtgate/schema/tracker.go [hunk 2]
```diff
@@ -134,20 +134,23 @@ func (t *Tracker) loadViews(conn queryservice.QueryService, target *querypb.Targ
...
+	// whenever a shard's primary tablet starts and sends the initial signal.
+	// This is needed clear out any stale view definitions.
 	t.clearKeyspaceViews(target.Keyspace)
-	t.updateViews(target.Keyspace, fvRes)
-	log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(fvRes))
 
+	var noOfViews int
+	err := conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil, func(schemaRes *querypb.GetSchemaResponse) error {
+		t.updateViews(target.Keyspace, schemaRes.TableDefinition)
+		noOfViews += len(schemaRes.TableDefinition)
```
Comment (ajm188) [inline go/vt/vtgate/schema/tracker.go:147]: this is an extreme nitpick (and especially if this was existing code please ignore it) -- to me, "no." only works as an abbreviation with the period included (which you obviously can't have here), how about `numViews` instead?

**Review by ajm188 — APPROVED** [review]
lgtm, I have one extremely annoying quibble

**Pull request merged**